### PR TITLE
Add support Linux/Windows Azure VM

### PR DESCRIPTION
**Pull request by vboulineau**
What does this PR do?
---------------------

Add support for Linux/Windows Azure VMs

Which scenarios this will impact?
-------------------

None, new scenarios

Motivation
----------

Additional Notes
----------------

The interface is low level at the moment and does not integrate with the `VM` type infrastructure due to on-going changes on this code in #13

### CODE REVIEW

File: azure/compute/vm.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+		AdminUsername: pulumi.String(adminUsername),
+		AdminPassword: windowsAdminPassword.Result,
+		CustomData:    userData,
+	}
+
+	if firstLogonCommand != nil {
+		windowsOsProfile.WindowsConfiguration = compute.WindowsConfigurationArgs{
+			AdditionalUnattendContent: compute.AdditionalUnattendContentArray{
+				compute.AdditionalUnattendContentArgs{
+					ComponentName: compute.ComponentNames_Microsoft_Windows_Shell_Setup,
```
Comment (pducolin) [inline azure/compute/vm.go:66]: **❓ question**
Is this to allow ssh connections?

File: azure/compute/vm.go [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+	})
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	vm, err := compute.NewVirtualMachine(e.Ctx, e.Namer.ResourceName(name), &compute.VirtualMachineArgs{
+		ResourceGroupName: pulumi.String(e.DefaultResourceGroup()),
+		VmName:            e.Namer.DisplayName(pulumi.String(name)),
+		HardwareProfile: compute.HardwareProfileArgs{
+			VmSize: pulumi.StringPtr(instanceType),
```
Comment (pducolin) [inline azure/compute/vm.go:124]: **💭 thought**
If `instanceType` is used to set the `VmSize`, should it be called `instanceSize` instead ?

File: azure/scenarios/vm/Pulumi.yaml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+name: vm
+runtime: go
+description: Agent VM
```
Comment (pducolin) [inline azure/scenarios/vm/Pulumi.yaml:3]: **🥜 nitpick**
```suggestion
description: Agent Windows VM
```

File: azure/scenarios/vm/vm/run.go [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+		return err
+	}
+
+	_, publicIP, nw, adminPassword, err := compute.NewWindowsInstance(env, "vm", compute.WindowsLatestURN(), env.DefaultInstanceType(), nil, nil)
+	if err != nil {
+		return err
+	}
+
+	ctx.Export("private-instance-ip", nw.IpConfigurations.Index(pulumi.Int(0)).PrivateIPAddress())
+	ctx.Export("public-instance-ip", publicIP.IpAddress)
```
Comment (pducolin) [inline azure/scenarios/vm/vm/run.go:21]: **❓ question**
Why do we need private and public address ?

File: azure/compute/vm.go [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+	linuxOsProfile := compute.OSProfileArgs{
+		ComputerName:  pulumi.String(name),
+		AdminUsername: pulumi.String(adminUsername),
+		LinuxConfiguration: compute.LinuxConfigurationArgs{
+			DisablePasswordAuthentication: pulumi.BoolPtr(true),
+			Ssh: compute.SshConfigurationArgs{
+				PublicKeys: compute.SshPublicKeyTypeArray{
+					compute.SshPublicKeyTypeArgs{
+						KeyData: sshPublicKey,
+						Path:    pulumi.StringPtr(fmt.Sprintf("/home/%s/.ssh/authorized_keys", adminUsername)),
```
Comment (pducolin) [inline azure/compute/vm.go:35]: **👏 praise**
Super nice !

**Review by pducolin — APPROVED** [review]

File: azure/compute/vm.go (hunk 2 above)
Comment (vboulineau) [reply azure/compute/vm.go:124]: So Azure calls it `VmSize` but it really is the `instanceType`, it's really the equivalent of the AWS `instanceType` and not just the # of CPUs.

File: azure/scenarios/vm/vm/run.go (hunk 4 above)
Comment (vboulineau) [reply azure/scenarios/vm/vm/run.go:21]: I have a bastion available in this network as well. Only SSH/RDP is allowed from external. If you need any other kind of interaction you will need to go through the bastion (and private ip)

File: azure/compute/vm.go (hunk 1 above)
Comment (vboulineau) [reply azure/compute/vm.go:66]: No, it's a way to run script on first-login. I **guess** it could be used to setup SSH, but I haven't verified that yet.

**Pull request merged**